WordPress/wordpress-develop
# Global Styles: Fix block spacing values being stripped for users without unfiltered_html capability

This PR backports https://github.com/WordPress/gutenberg/pull/46388 and https://github.com/WordPress/gutenberg/pull/46712 which resolved https://github.com/WordPress/gutenberg/issues/45520 in Gutenberg.

To recap: for users without the `unfiltered_html` capability (e.g. an Admin user within a multi site WordPress instance), when saving custom block spacing values or content/wide layout sizes within global styles, the value is stripped on save. The fix is to ensure that indirect properties (that is, Theme JSON properties that are not directly output via `compute_style_properties` within `remove_insecure_styles` and `remove_insecure_settings`) are also allowed.

For more context, please see: https://github.com/WordPress/gutenberg/pull/46388 

Trac ticket: https://core.trac.wordpress.org/ticket/57321

## Testing instructions

1. To quickly test, add the following somewhere (e.g. at the bottom of `src/wp-includes/default-filters.php`): `add_action( 'init', 'kses_init_filters' );` — this switches on the kses filters used when a user does not have the `unfiltered_html` capability.
2. With a blocks theme active, go to global styles within the site editor, and adjust block spacing within layout. When you go to save a value, without this PR, the value will revert to what it was when you originally opened the site editor.
3. With this PR applied, the value should save correctly.
4. You can repeat this process at the block level in global styles, with a block that supports block spacing/ block gap, e.g. Social Icons/Links.
5. Also, you can repeat this process with the content and wide layout sizes within global styles. With this PR applied, updates to content / wide layout sizes should also now be respected when saved.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
